### PR TITLE
Add configuration class

### PR DESCRIPTION
**Pull request by sanyakosivchuk**
- [x] Extract configuration class 
- [x] Update with RubyAI::Configuration.new("api_key")
- [x] Extract headers in HTTP headers. 
- [x] Update Readme

### CODE REVIEW

**Comment by alexshapalov** [comment]
```
class Configuration
  # attr_reader :api_key

  attr_accessor :model, :messages, :temperature, :api_version, :api_key

  BASE_URL = "https://api.openai.com/v1/chat/completions"

  DEFAULT_API_VERSION = "v1"

  MODELS = {
    "gpt-4" => "gpt-4",
    "gpt-4-0314" => "gpt-4-0314",
    "gpt-4-32k" => "gpt-4-32k",
    "gpt-3.5-turbo" => "gpt-3.5-turbo",
    "gpt-3.5-turbo-0301" => "gpt-3.5-turbo-0301",
    "text-davinci-003" => "text-davinci-003"
  }

  DEFAULT_MODEL = "gpt-3.5-turbo"

  def initialize(api_key, model = DEFAULT_MODEL)
    @api_key = nil
    @model = model
    @api_version = DEFAULT_API_VERSION
  end
end

conf = Configuration.new("api_key")

p conf.model
```



```
require_relative "configuration"

# conf = Configuration.new("api_key")

# p conf.model

module RubyAI
  # class << self
  #   attr_writer :configuration
  # end

  def self.configuration
    @configuration ||= Configuration.new("api_key")
  end

  def self.call
    @configuration
  end

  def self.configure
    yield(configuration)
  end
end


p RubyAI.configuration

# Configure RubyAI

# p RubyAI::Configuration.new("api_key")

RubyAI.configure do |config|
  p config.api_key = 'new value for api_key 1'
  p config.model = 'new value for model 2'
end
```

**Comment by sanyakosivchuk** [comment]
both me and @bodyyyya worked on Extract configuration class commit

**Comment by sanyakosivchuk** [comment]
for now it works with
```
configuration = RubyAI::Configuration.new("YOUR_API_KEY", "Who is the best chess player in history?")
client = RubyAI::Client.new(configuration)
result = client.call
puts result.dig("choices", 0, "message", "content")
```
and
```
api_key = "YOUR_API_KEY"
messages = "Who is the best chess player in history?"
result = RubyAI::Client.new(api_key, messages).call
puts result.dig("choices", 0, "message", "content")
```
ill update readme later

File: lib/rubyai/client.rb [hunk 1]
```diff
@@ -1,55 +1,41 @@
-require 'faraday'
-require 'json'
-
+require 'faraday'
```
Comment (alexshapalov) [inline lib/rubyai/client.rb:1]: Do we need to require it here? json and faraday

**Review by alexshapalov — CHANGES_REQUESTED** [review]

File: lib/rubyai/client.rb [hunk 2]
```diff
@@ -1,55 +1,41 @@
...
-    attr_accessor :api_key, :model, :messages, :temperature
-
     def call
       response = connection.post do |req|
-        req.url BASE_URL
-        req.headers['Content-Type'] = 'application/json'
-        req.headers['Authorization'] = "Bearer #{@api_key}"
-        req.body = body.to_json
+        req.url RubyAI::Configuration::BASE_URL
+        req.headers.merge!(RubyAI::HTTP.build_headers(@configuration.api_key))
```
Comment (alexshapalov) [inline lib/rubyai/client.rb:17]: if we have **attr_reader :configuration** we can use everywhere just **configuration**

File: lib/rubyai/client.rb [hunk 3]
```diff
@@ -1,55 +1,41 @@
...
 
-    private
-
-    def body
-      {
-        'model': MODELS[@model],
-        'messages': [{"role": "user", "content": @messages}],
-        'temperature': @temperature
-      }
+    def apply_options(options)
```
Comment (alexshapalov) [inline lib/rubyai/client.rb:24]: Where do we use this method?

File: lib/rubyai/http.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require_relative 'configuration'
+
+module RubyAI
+  module HTTP
+    module_function
```
Comment (alexshapalov) [inline lib/rubyai/http.rb:5]: Do not use module_function; it is better to use **extend self**

**Review by alexshapalov — CHANGES_REQUESTED** [review]

**Review by alexshapalov — APPROVED** [review]

**Pull request merged**